### PR TITLE
tool/exchange_wrapper_coverage: fix regression and implement reflection 

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Came across this in another branch on implementing supplementary changes. Added reflection so as to dynamically scale with GCT exchange interface. Can build more on to this later. 

Note: As reflection uses zero values, I also implemented checks for wrapper functions before nil pointer interactions as this was panic at the disco.

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [hope so] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cmd/exchange_wrapper_coverage/main.go [hunk 1]
```diff
@@ -73,149 +78,42 @@ func main() {
...
-	if errors.Is(err, common.ErrNotYetImplemented) {
-		funcs = append(funcs, "GetAccountInfo")
-	}
-
-	_, err = e.GetRecentTrades(context.TODO(), p, assetType)
-	if errors.Is(err, common.ErrNotYetImplemented) {
-		funcs = append(funcs, "GetRecentTrades")
-	}
+// errValue extract reflection type
+var errValue = reflect.TypeOf(errors.New(""))
```
Comment (gloriousCode) [inline cmd/exchange_wrapper_coverage/main.go:82]: Can localise to the function

File: exchanges/bitmex/bitmex_test.go [hunk 2]
```diff
@@ -57,8 +58,15 @@ func TestMain(m *testing.M) {
 
 func TestStart(t *testing.T) {
 	t.Parallel()
+	err := b.Start(nil)
```
Comment (gloriousCode) [inline exchanges/bitmex/bitmex_test.go:61]: Let's get this on all exchanges 🎉

File: config/config.go [hunk 3]
```diff
@@ -29,6 +29,9 @@ import (
 	"github.com/thrasher-corp/gocryptotrader/portfolio/banking"
 )
 
+// ErrExchangeConfigIsNil defines an error when the config is nil
+var ErrExchangeConfigIsNil = errors.New("exchange config is nil")
```
Comment (gloriousCode) [inline config/config.go:33]: Can be unexported

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Good stuff! 
# :chart_with_upwards_trend:

File: exchanges/kraken/kraken_test.go [hunk 4]
```diff
@@ -61,6 +62,21 @@ func TestMain(m *testing.M) {
 }
 
 // TestGetServerTime API endpoint test
```
Comment (gloriousCode) [inline exchanges/kraken/kraken_test.go:64]: Please RM

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Just one outrageous issue from me now

**Review by gloriousCode — APPROVED** [review]
Nice work Shazbert! Thanks for making those changes

File: cmd/exchange_template/wrapper_file.tmpl [hunk 5]
```diff
@@ -171,12 +177,16 @@ func ({{.Variable}} *{{.CapitalName}}) Setup(exch *config.Exchange) error {
 }
 
 // Start starts the {{.CapitalName}} go routine
-func ({{.Variable}} *{{.CapitalName}}) Start(wg *sync.WaitGroup) {
+func ({{.Variable}} *{{.CapitalName}}) Start(wg *sync.WaitGroup) error {
+	if wg == nil {
+		return fmt.Errorf("%T %w", wg, common.ErrNilPointer)
```
Comment (thrasher-) [inline cmd/exchange_template/wrapper_file.tmpl:182]: Need to add "fmt" to the import list

**Review by thrasher- — CHANGES_REQUESTED** [review]
Just one nit, the coverage tool works well with the dynamic method update!

image

File: cmd/exchange_template/wrapper_file.tmpl (hunk 5 above)
Comment (shazbert) [reply cmd/exchange_template/wrapper_file.tmpl:182]: 🦅

**Review by thrasher- — APPROVED** [review]
tACK, thanks for addressing the nit 🚀 !

**Pull request merged**